osquery/tables/networking/linux/iptables.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <sstream>
+
+#include <arpa/inet.h>
+#include <libiptc/libiptc.h>
+
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string/trim.hpp>
+
+
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+
+namespace osquery {
+namespace tables {
+
+const std::string kLinuxIpTablesNames = "/proc/net/ip_tables_names";
+const char MAP[] = {'0','1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
+const int HIGH_BITS = 4;
+const int LOW_BITS = 15;
+
+QueryData getIptablesRules(const std::string& content) {
+  QueryData results;
+
+  for (auto& line : split(content, "\n")) {
+    if (line.size() == 0) {
+      continue;
+    }
+
+    // Inline trim each line.
+    boost::trim(line);
+
+    Row r;
+
+    r["filter_name"] = TEXT(line);
+
+    struct iptc_handle *h;
+
+    // Initialize the access to iptc
+    h = (struct iptc_handle*) iptc_init(line.c_str());
```

Comment:
you could remove the out-of-band instantiation and just do:

```cpp
struct iptc_handle *h = (struct iptc_handle*) iptc_init(line.c_str());
```

or even:

```cpp
auto h = (struct iptc_handle*) iptc_init(line.c_str());
```

also, while you're there, can you use a >= 3 character variable name?